docs/ECOS.md

```diff
@@ -0,0 +1,141 @@
+# Edge Containers
+
+## Introduction
+
+While EVE provides a lot of sophisticated device management functionality, at the end of the day its success depends on a diverse and growing ecosystem of applications that it can support. EVE is *not* a traditional operating system in that it doesn't support applications represented as executable files and [POSIX processes](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_295). Instead EVE expects its applications to be:
+
+* [Traditional Virtual Machines](https://en.wikipedia.org/wiki/Virtual_machine)
+* [Unikernels](http://unikernel.org/)
+* [Docker/OCI container](https://www.opencontainers.org)
+
+Edge Container is a novel concept that allows an effective packaging and lifecycle management of all these types of applications and it consists of two parts:
+
+* Edge Container Image (ECI) - an self-contained, binary representation of an Edge Container
+* Edge Container Object (ECO) - a live copy of an Edge Container
+
+ECOs are created by users binding a single ECI to a set of hardware and software resources (e.g. memory, CPU, I/O and networks) and we sometimes use the term ECO to refer to both -- an overall concept of an Edge Container (because it is an Edge COntainer) and a live copy. It will be fair to say that the primary purpose of EVE as an engine is to provide the best resource abstraction and an execution environment to a set of ECOs.
```

Comment:
This paragraph confused me in several ways;

* "binding a single ECI to a set of hardware and software" - what do we mean by "binding"? 
* "we sometimes use the term ECO to refer to both" - both of what? Isn't the "binding" (if I understood it correctly; not sure I did) just the running? 
* "sometimes use the term... to refer to both" - this leaves readers confused as to when we will do what. We should be clearer.

Separately:

* "It will be fair to say" - language is a little weak for writing, works well in a verbal presentation. Can we be more assertive? "The primary purpose of EVE is to provide the best resource abstraction and execution environment for ECOs." We are confident, let's just state it.
* when we say "resource abstraction", do we mean allowing ECO designers to just say, "I want ½ CPU and 500MB of memory for this workload to run", and let them not worry about it? If so, it makes sense, but weakens the sentence. "The primary purpose of EVE is to provide the best execution environment for Edge Containers." That almost could be a summary to the whole purpose of EVE. We can describe resource abstraction and such elsewhere.

Reply:
Good points. Rewrote it to make it clearer.